### PR TITLE
change-default-produces-mediatype

### PR DESCRIPTION
**Pull request by dvddhln**
Suggestion for the default produces to be `MediaType.WILDCARD` .

### CODE REVIEW

File: implementation/src/main/java/io/smallrye/restclient/RestClientBuilderImpl.java [hunk 1]
```diff
@@ -133,7 +133,7 @@ public RestClientBuilder baseUrl(URL url) {
                 .proxyBuilder(aClass)
                 .classloader(classLoader)
                 .defaultConsumes(MediaType.TEXT_PLAIN)
```
Comment (kenfinnigan) [inline implementation/src/main/java/io/smallrye/restclient/RestClientBuilderImpl.java]: Should the default `@Consume` also change?

Comment (dvddhln) [reply implementation/src/main/java/io/smallrye/restclient/RestClientBuilderImpl.java]: Not sure here, is the default consumes really needed as it produces a Content-type: */* ? Perhaps its better not to send it at all I mean.

Comment (kenfinnigan) [reply implementation/src/main/java/io/smallrye/restclient/RestClientBuilderImpl.java]: Ah, you have a point. Maybe remove that line and see if the TCK passes

Comment (dvddhln) [reply implementation/src/main/java/io/smallrye/restclient/RestClientBuilderImpl.java]: Didn't pass it seems.

 According to RFC 7231 

> If a Content-Type header field is not present, the recipient MAY either assume a media type of "application/octet-stream ([RFC2046], Section 4.5.1) or examine the data to determine its type." 

Should I try `MediaType.APPLICATION_OCTET_STREAM` for the `defaultConsumes`?

Comment (kenfinnigan) [reply implementation/src/main/java/io/smallrye/restclient/RestClientBuilderImpl.java]: Sure

Comment (dvddhln) [reply implementation/src/main/java/io/smallrye/restclient/RestClientBuilderImpl.java]: Seems to be in better shape now!

**Pull request merged**